### PR TITLE
checkcommits: allow subsystem to overide need for "Fixes #XXX"

### LINKED ISSUE #820: checkcommits: allow a release PR to not require a "Fixes" comment
Add a new option to `checkcommits` to permit a release PR to not need to specify a `Fixes #XXX` comment. This will simplify the release process and allow it to be fully automated.

- See: https://github.com/clearcontainers/packaging/pull/229

### PR DESCRIPTION
**Pull request by jodh-intel**
Add a new `--ignore-fixes-for-subsystem=` option that if specified will
disable the effect of `--need-fixes` for those branches containing atleast one
commit with the specified subsystem.

The most useful scenario for this option being to allow a release PR
(subsystem "`release: ...`") to not require an issue to be raised for it.

Fixes #820.

Signed-off-by: James O. D. Hunt <james.o.hunt@intel.com>

### CODE REVIEW

File: cmd/checkcommits/checkcommits.go [hunk 1]
```diff
@@ -102,19 +112,28 @@ func checkCommitSubject(config *CommitConfig, commit, subject string) error {
...
+
+	var subsystem string
+
+	if matches == nil || len(matches) != 2 {
+		return fmt.Errorf("Commit %v: Failed to find subsystem in subject: %q", commit, subject)
 	}
 
+	// match 1 is the entire matching string
+	// match 2 is the subsystem name (without the colon)
+	subsystem = matches[1]
```
Comment (jcvenegas) [inline cmd/checkcommits/checkcommits.go]: accorinding to the comment, dont you need matches[2]?

Comment (jodh-intel) [reply cmd/checkcommits/checkcommits.go]: Good question. Those comments were supposed to refer to "1st" and "2nd", but I've changed it to make it clearer. Thanks!

**Comment by devimc** [comment]
would be nice to have support for ```fixes ORG/PROJECT#NUM```
for example ```fixes clearcontainers/agent#198``` to make refence to a issue that is not part of this repository but is part of the same organization

**Comment by devimc** [comment]
lgtm

[Approved with PullApprove](https://pullapprove.com/clearcontainers/tests/pull-request/861/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=clearcontainers/tests) [Approved with PullApprove](https://pullapprove.com/clearcontainers/tests/pull-request/861/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=clearcontainers/tests)

**Pull request merged**